tests/test_constellation.py

```diff
@@ -64,6 +69,60 @@ def test_api_configured():
     obj.stop(kill=True)
 
 
+def test_proxy_configured_self_signed():
+    cfg = MontaguConfig("config/basic")
+    obj = MontaguConstellation(cfg)
+
+    obj.start()
+
+    api = get_container(cfg, "proxy")
+    cert = docker_util.string_from_container(api, "/etc/montagu/proxy/certificate.pem")
+    key = docker_util.string_from_container(api, "/etc/montagu/proxy/ssl_key.pem")
+    param = docker_util.string_from_container(api, "/etc/montagu/proxy/dhparam.pem")
+    assert cert is not None
+    assert key is not None
+    assert param is not None
+
+    res = http_get("https://localhost")
+    assert "Montagu" in res
+
+    obj.stop(kill=True)
+
+
+def test_proxy_configured_ssl():
+    cfg = MontaguConfig("config/complete")
+    obj = MontaguConstellation(cfg)
+
+    obj.start()
+
+    api = get_container(cfg, "proxy")
+    cert = docker_util.string_from_container(api, "/etc/montagu/proxy/certificate.pem")
+    key = docker_util.string_from_container(api, "/etc/montagu/proxy/ssl_key.pem")
+    param = docker_util.string_from_container(api, "/etc/montagu/proxy/dhparam.pem")
+    assert cert == "cert"
+    assert key == "k3y"
+    assert param == "param"
+
+    obj.stop(kill=True)
+
+
 def get_container(cfg, name):
     cl = docker.client.from_env()
     return cl.containers.get(f"{cfg.container_prefix}-{cfg.containers[name]}")
+
+
+# Because we wait for a go signal to come up, we might not be able to
+# make the request right away:
+def http_get(url, retries=5, poll=0.5):
+    ctx = ssl.create_default_context()
+    ctx.check_hostname = False
+    ctx.verify_mode = ssl.CERT_NONE
+    for _i in range(retries):
+        try:
+            r = urllib.request.urlopen(url, context=ctx)  # noqa
```

Comment:
this method is a security risk apparently (https://stackoverflow.com/questions/48779202/audit-url-open-for-permitted-schemes-allowing-use-of-file-or-custom-schemes) but since we're the only ones calling it, doesn't seem to matter